File: packages/pluto/src/clients/shared/router.ts
import { utils } from "@plutolang/base";
import { IRouterClient, Router, RouterOptions } from "../../router";

export class RouterClient implements IRouterClient {
  private readonly id: string;

  constructor(name: string, opts?: RouterOptions) {
    this.id = utils.genResourceId(Router.fqn, name);
    opts;
  }

  url(): string {
    return utils.getEnvValForProperty(this.id, "url");
  }
}

File: packages/pluto-py/pluto_client/clients/shared/router.py
from typing import Optional
from pluto_base.utils import gen_resource_id, get_env_val_for_property
from ...router import IRouterClient, Router, RouterOptions


class RouterClient(IRouterClient):
    def __init__(self, name: str, opts: Optional[RouterOptions] = None):
        self.__id = gen_resource_id(Router.fqn, name)

    def url(self) -> str:
        return get_env_val_for_property(self.__id, "url")
